### PR TITLE
Initial bemenu module

### PR DESCRIPTION
**Pull request by Lyndeno**
Initial bemenu config with sane defaults. Sets the options via BEMENU_OPTS so they can also be overridden.

Includes optional settings such as ```fontSize``` and ```alternate```. These options can be set to adjust certain settings in bemenu.

### CODE REVIEW

File: modules/bemenu.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{pkgs, config, lib, ... }:
+
+with config.lib.stylix.colors.withHashtag;
+with config.stylix.fonts;
+{
+  options.stylix.targets.bemenu = {
+    enable = config.lib.stylix.mkEnableTarget "bemenu" true;
+    fontSize = lib.mkOption { type = lib.types.str; default = ""; }; # optional argument
```
Comment (danth) [inline modules/bemenu.nix:8]: This should probably be `int` type:

```suggestion
    fontSize = lib.mkOption {
      type = with lib.types; nullOr int;
      default = null;
    };
```

Then convert it to the appropriate string where it's needed:

```nix
${lib.optionalString (fontSize != null) (builtins.toString fontSize)}
```

It's a bit more wordy, but it catches potential mistakes like setting `"bold"` as the font size.

File: modules/bemenu.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    alternate = lib.mkOption { type = lib.types.bool; default = false; };
+  };
+
+  config.home-manager.sharedModules = lib.mkIf config.stylix.targets.mako.enable [{
```
Comment (danth) [inline modules/bemenu.nix:12]: Almost missed this one!

```suggestion
  config.home-manager.sharedModules = lib.mkIf config.stylix.targets.bemenu.enable [{
```

**Review by danth — CHANGES_REQUESTED** [review]
Documentation didn't like the settings with no description - but I've sorted that out now. Rebasing this on master should make the checks pass.

**Comment by Lyndeno** [comment]
Made some updates including your changes

**Review by danth — APPROVED** [review]
Thanks!

**Pull request merged**